ui/src/app/pages/workspace/workspace-share.tsx

```diff
@@ -277,15 +297,24 @@ export const WorkspaceShare = fp.flow(withUserProfile())(
         email: user.email,
         role: WorkspaceAccessLevel.READER,
       };
+
+      const userRoles = fp.sortBy(
```

Comment:
userRoles is what is displayed in the modal: simply add the collab.

userRolesToChange is the changeset we use in the share request.  Most likely, it's not there yet.  But we may have already added this user to userRolesToChange as an explicit deletion, so we need to filter this out first.